### ISSUE TITLE
Test failling LogStash::Instrument::PeriodicPoller::JVM load average on linux when an exception occur reading the file doesn't raise an exception

### ISSUE THREAD

**Issue by ph** [elastic/logstash#6354]
```
  1) LogStash::Instrument::PeriodicPoller::JVM load average on linux when an exception occur reading the file doesn't raise an exception
     Failure/Error: expect(::File).to receive(:read).with("/proc/loadavg").and_raise("Didnt work out so well")
       (<File (class)>).read("/proc/loadavg")
           expected: 1 time with arguments: ("/proc/loadavg")
           received: 2 times with arguments: ("/proc/loadavg")
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/logstash-core/spec/logstash/instrument/periodic_poller/jvm_spec.rb:47:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/rakelib/test.rake:49:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/task.rb:248:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/task.rb:243:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/task.rb:187:in `invoke_with_call_chain'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/task.rb:180:in `invoke_with_call_chain'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/task.rb:173:in `invoke'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:152:in `invoke_task'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:108:in `top_level'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:108:in `top_level'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:117:in `run_with_threads'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:102:in `top_level'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:80:in `run'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:178:in `standard_exception_handling'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/lib/rake/application.rb:77:in `run'
     # /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/vendor/bundle/jruby/1.9/gems/rake-11.3.0/exe/rake:27:in `(root)'

Finished in 2 minutes 37.4 seconds (files took 5.29 seconds to load)
2228 examples, 1 failure

Failed examples:

rspec /var/lib/jenkins/workspace/elastic+logstash+5.1+multijob-os-compatibility/os/opensuse/logstash-core/spec/logstash/instrument/periodic_poller/jvm_spec.rb:50 # LogStash::Instrument::PeriodicPoller::JVM load average on linux when an exception occur reading the file doesn't raise an exception

Randomized with seed 1024

**Comment by ph** [comment]
I've tried reproducing it on linux and mac without luck so far with and without  the same seed. I believe its not related to the order of execution of the tests. The other thing is maybe we have another thread in the background trying to get metric.

maybe we should stop collecting metric when we run the tests scenario and only do it when required. This is also another argument to refactor the metric collection so they don't mix with the actual logic of the classes.

**Comment by ph** [comment]
I have fixed it in #6886
